.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -14,3 +14,12 @@ Anything else we should know when reviewing?
 
 Write here in detail how you have tested your changes
 and instructions on how this should be tested in QA.
+
+### Checklist
+<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->
+
+- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
```

Comment:
This is already enforced by CI so maybe we don't need it here. IMO the checklist should be for things that are not easy to check automatically

Reply:
This was a suggestion from Agent Core and I think it can be useful for external contributors, I have no strong opinion either way.

Reply:
I see, it could be useful for external contributors if they don't know how to interact with CircleCI so let's keep it